tests/test_parser/test_causaleffect.py

```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+
+"""Test the causaleffect probability expression parser."""
+
+import unittest
+
+from y0.dsl import C, D, P, Sum, Variable
+from y0.parser import parse_causaleffect
+
+U_1 = Variable('U_1')
+
+
+class TestCausaleffectGrammar(unittest.TestCase):
+    """Tests for parsing causaleffect probability expressions."""
+
+    def test_parse(self):
+        """Test the high-level parser API."""
+        for expr, s in [
+            (Sum[U_1, C](P(D | U_1 | C) * P(C) * P(U_1)), "\\sum_{U_{1},C}P(D|u_{1},C)P(C)P(U_{1})"),
```

Comment:
@djinnome I'll need more examples (or even better, documentation) before I can write the actual parser for the causaleffect terms. For now, I will assume it's pretty close to craig-like notation except for some of the slashes, squigglies, and underscores